### ISSUE TITLE
No insertRule cache?

### ISSUE THREAD

**Issue by steadicat** [streamich/nano-css#214]
I’ve been using the `virtual` add-on mainly because it has the side effect of not inserting new rules on every render and infinitely growing the stylesheet. Since I noticed that `virtual` has [issues with server rendering](https://github.com/streamich/nano-css/issues/213), I tried disabling it, and I was surprised to see that by default, Nano does not keep a cache rules previously injected in the stylesheet.

If you inline rule definitions in your React render methods, new rules get added on every render, even if they are identical.

Here's a screenshots from logging before `insertRule`. I've got 600 rules right after load, and  identical rules are injected all the time:

<img width="280" alt="image" src="https://user-images.githubusercontent.com/6031/49823351-502c3000-fd34-11e8-97bf-e67b8f1dada9.png">

With a quick custom solution I wrote that does caching, the number of rules on the same page plateaus somewhere around 80:

<img width="385" alt="image" src="https://user-images.githubusercontent.com/6031/49823710-317a6900-fd35-11e8-8355-1d7e7a3f03d9.png">

I think Emotion and most other CSS libraries optimize this part. It seems like a relatively simple optimization. Any reason not to do it?

**Comment by streamich** [comment]
`virtual` addon is supposed to cache all atoms internally, here is where it is supposed to take a class name from cache:

- https://github.com/streamich/nano-css/blob/master/addon/virtual.js#L55

But from your post I understand that there must be a bug.

**Comment by steadicat** [comment]
You're right that `virtual` addon caches correctly. I was talking about what happens without it, i.e. the default behavior.

**Comment by streamich** [comment]
OK, then what you basically mean is that `rule` addon is not caching. I did not think it would need caching. There could be another addon that adds caching:

```js
nano.put = momoized(nano.put);
```
